pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -517,3 +568,16 @@ func (s *NodeController) getCondition(node *api.Node, conditionType api.NodeCond
 	}
 	return nil
 }
+
+// latestConditionTime returns the latest condition timestamp for the node, regardless of condition status.
+// If nothing matches, the node creation timestamp will be returned.
+func (s *NodeController) latestConditionTime(node *api.Node, conditionType api.NodeConditionType) util.Time {
+	readyTime := node.ObjectMeta.CreationTimestamp
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == conditionType &&
+			condition.LastProbeTime.After(readyTime.Time) {
```

Comment:
IIUC, in both the case of syncNodeStatus==true and syncNodeStatus==false, LastProbeTime tells you the last time you got a response from Kubelet. Is that correct? If so, could you please change LastProbeTime to LastStatusTime or LastConditionTime? LastProbeTime implies the last time you sent a probe to the Kubelet, not the last time you heard from the Kubelet. (I can file a separate issue for it, but please let me know if my understanding is correct).